lib/nethealth/nethealth_test.go

```diff
@@ -0,0 +1,343 @@
+		assert.Equal(t, tt.expected, server.peers, tt.description)
+	}
+
+}
+
+func TestResyncPods(t *testing.T) {
+	server := testServer(t)
+	nodes := []*v1.Node{
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "node-1",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "peer-1",
+			},
+		},
+		{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "peer-2",
+			},
+		},
+	}
+
+	cases := []struct {
+		add            []*v1.Pod
+		rm             []string
+		expectedPeers  map[string]*peer
+		expectedLookup map[string]string
+		description    string
+	}{
+		{
+			description: "ignore our own node",
+			add: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "pod-node-1",
+						Labels: map[string]string{
+							"k8s-app": "nethealth",
+						},
+					},
+					Spec: v1.PodSpec{
+						NodeName: "node-1",
+					},
+					Status: v1.PodStatus{
+						PodIP: "1.1.1.1",
+					},
+				},
+			},
+			expectedPeers: map[string]*peer{
+				"peer-1": {
+					name:             "peer-1",
+					lastStatusChange: server.clock.Now(),
+					addr:             &net.IPAddr{},
+				},
+				"peer-2": {
+					name:             "peer-2",
+					lastStatusChange: server.clock.Now(),
+					addr:             &net.IPAddr{},
+				},
+			},
+			expectedLookup: map[string]string{},
+		},
+		{
+			description: "new pod for peer-1",
+			add: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "pod-peer-1",
+						Labels: map[string]string{
+							"k8s-app": "nethealth",
+						},
+					},
+					Spec: v1.PodSpec{
+						NodeName: "peer-1",
+					},
+					Status: v1.PodStatus{
+						PodIP: "1.1.1.1",
+					},
+				},
+			},
+			expectedPeers: map[string]*peer{
+				"peer-1": {
+					name:             "peer-1",
+					lastStatusChange: server.clock.Now(),
+					addr: &net.IPAddr{
+						IP: net.ParseIP("1.1.1.1"),
+					},
+				},
+				"peer-2": {
+					name:             "peer-2",
+					lastStatusChange: server.clock.Now(),
+					addr:             &net.IPAddr{},
+				},
+			},
+			expectedLookup: map[string]string{
+				"1.1.1.1": "peer-1",
+			},
+		},
+		{
+			description: "pod that doesn't have a node set",
+			add: []*v1.Pod{
+				{
+					ObjectMeta: metav1.ObjectMeta{
+						Name: "pod-unknown-peer-1",
+						Labels: map[string]string{
+							"k8s-app": "nethealth",
+						},
+					},
+					Spec: v1.PodSpec{
+						NodeName: "",
+					},
+					Status: v1.PodStatus{
+						PodIP: "1.1.1.1",
+					},
+				},
+			},
+			expectedPeers: map[string]*peer{
+				"peer-1": {
+					name:             "peer-1",
+					lastStatusChange: server.clock.Now(),
+					addr: &net.IPAddr{
+						IP: net.ParseIP("1.1.1.1"),
+					},
+				},
+				"peer-2": {
+					name:             "peer-2",
+					lastStatusChange: server.clock.Now(),
+					addr:             &net.IPAddr{},
+				},
+			},
+			expectedLookup: map[string]string{
+				"1.1.1.1": "peer-1",
+			},
+		},
+		{
+			description: "pod that doesn't have an equivelant node object",
```

Comment:
Same here - not sure which changes (or lack thereof) to expect...

Reply:
This is a pod that is running on a node the process doesn't know about. So if we haven't collected / cached a kubernetes object, but we have a pod that points to a node we don't know about, this is meant to validate the behaviour that this is properly ignored.